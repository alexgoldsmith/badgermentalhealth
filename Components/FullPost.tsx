import React, { useEffect, useState } from 'react'
import { StyleSheet, Text, View, TouchableOpacity, Image } from 'react-native'
import deletePost from '../Hooks/deletePost';
import getStorageData from '../Hooks/getStorageData';
import updatePost from '../Hooks/updatePost';
import { Post } from '../Interfaces/Interfaces'
import getStyles from '../Styling/Styling'

const styles = getStyles();

type propType = {
    post: Post,
    refresh: Function,
}

const FullPost: React.FC<propType> = ({post, refresh}) => {
    const [likes, setLikes] = useState<number>(post.Likes)
    const [user, setUser] = useState<string>('')
    
    useEffect(() => {
        loadUser()
    }, [])

    const loadUser = async () => {
        const data = await getStorageData('user')
        setUser(data.email)
    }
    
    const increaseLikes = () => {
        setLikes(likes + 1)
        updateLikes(likes + 1)
    }
    
    const updateLikes = (likes) => {
        post.Likes = likes
        updatePost(post);
    }

    return (
        <View
            style={[styles.postCard, {width: '95%', padding: 20}]}
        >
            <Text style={{fontSize: 24, fontWeight: 'bold'}}>{post.Title}</Text>
            <Text style={{fontSize: 14, marginBottom: 15}}>{post.Text}</Text>
            <TouchableOpacity 
                    style={{display: 'flex', flexDirection: 'row', alignItems: 'center', width: 50, marginRight: 30}}
                    onPress={increaseLikes}
            >
                <View style={{marginTop: 1, display: 'flex', alignItems: 'center', flexDirection: 'row'}}>
                    <Image style={styles.like_image} source={require('../assets/like.webp')} /> 
                    <Text style={{fontSize: 20}}> {likes}  </Text>
                </View>
                {(post.Author === user || post.Author === 'admin@admin.com')?
                    <TouchableOpacity 
                        style={[styles.loginCard, styles.redBorder, {backgroundColor: 'white', height: 40, minWidth: 150, width: 150, margin: 15, marginLeft: 100}]}
                        onPress={() => {deletePost(post); refresh()}}
                    >
                        <Text style={{fontSize: 16, color: '#8f0000', textDecorationLine: 'underline', margin: 0}}>Delete</Text>
                    </TouchableOpacity>
                    : <></>
                }
            </TouchableOpacity>
        </View>
    )
}

export default FullPost
