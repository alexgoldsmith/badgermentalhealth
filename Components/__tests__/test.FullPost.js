// https://jestjs.io/docs/snapshot-testing#snapshot-testing-with-jest
// components/__tests__/Hello.tsx
import React from 'react';
import renderer from 'react-test-renderer';
import FullPost from '../FullPost';

const mockPost = {
    _id: 3484,
    Title: "Test Title",
    Author: "Test Author",
    Date: new Date(2021, 23, 7),
    Likes: 5,
    NumReplis: 0,
}

it('renders correctly with defaults', () => {
  const component = renderer
    .create(<FullPost post={mockPost}/>)
    .toJSON();
  expect(component).toMatchSnapshot();
});