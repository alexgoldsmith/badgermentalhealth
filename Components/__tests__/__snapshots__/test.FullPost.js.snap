// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 10,
      "display": "flex",
      "justifyContent": "center",
      "margin": 20,
      "opacity": 1,
      "padding": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    Test Title
  </Text>
  <Text
    style={
      Object {
        "fontSize": 14,
      }
    }
  />
  <Text>
    Author: 
    Test Author
  </Text>
  <Text>
    Likes: 
    5
     
  </Text>
</View>
`;
